File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode start = new ListNode(), cur = start;
        while (cur != null) {
            if (l1 == null && l2 == null) {
                cur = null;
                break;
            }
            if (l1 == null) {
                cur.next = l2;
                break;
            }
            else if (l2 == null) {
                cur.next = l1;
                break;
            }
            else {
                if (l1.val > l2.val) {
                    cur.next = l2;
                    l2 = l2.next;
                } else {
                    cur.next = l1;
                    l1 = l1.next;
                }
            }
            cur = cur.next;
        }
        return start.next;
    }
}

File: Data Structures & Algorithms/meeting-schedule-ii/submission-1.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:
        ans = 0
        start_end = []
        for interval in intervals:
            start_end.append([interval.start, 0])
            start_end.append([interval.end, 1])
        
        # 时间相同时，把end排在前面
        start_end.sort(key=lambda x:(x[0],-x[1]))

        cur = 0
        for se in start_end:
            if se[1] == 0:
                cur += 1
                ans = max(ans, cur)
            else:
                cur -= 1
        
        return ans